File: WorkShop Five/Tree.py
class Node:
    def __init__(self, data):
        self.data = data
        self.children = []
    
    def add_child(self, child):
        self.children.append(child)
        return child

def print_tree(node, level=0, prefix="└── "):
    # Print the current node with proper indentation
    print("    " * level + prefix + node.data)

    # Recursively print all children
    for child in node.children:
        print_tree(child, level + 1)

# Create the tree structure
def build_store_tree():
    # Create the root node
    store = Node("Store")
    
    # Create main categories
    electronics = Node("Electronics")
    clothing = Node("Clothing")
    
    # Add main categories to store
    store.add_child(electronics)
    store.add_child(clothing)
    
    # Create subcategories for Electronics
    phones = Node("Phones")
    computers = Node("Computers")
    
    # Add subcategories to Electronics
    electronics.add_child(phones)
    electronics.add_child(computers)
    
    # Create subcategories for Clothing
    men = Node("Men")
    women = Node("Women")
    
    # Add subcategories to Clothing
    clothing.add_child(men)
    clothing.add_child(women)
    
    return store

# Main function to run the program
def main():
    # Build the tree
    store_tree = build_store_tree()
    
    # Print the tree structure
    print("Store Product Categories:")
    print_tree(store_tree)

if __name__ == "__main__":
    main()

File: WorkShop Five/Tree.js
class Node {
  constructor(data) {
    this.data = data;
    this.children = [];
  }

  addChild(child) {
    this.children.push(child);
    return child;
  }
}

function printTree(node, level = 0, prefix = "└── ") {
  // Print the current node with proper indentation
  console.log("    ".repeat(level) + prefix + node.data);
  
  // Recursively print all children
  for (const child of node.children) {
    printTree(child, level + 1);
  }
}

// Create the tree structure
function buildStoreTree() {
  // Create the root node
  const store = new Node("Store");
  
  // Create main categories
  const electronics = new Node("Electronics");
  const clothing = new Node("Clothing");
  
  // Add main categories to store
  store.addChild(electronics);
  store.addChild(clothing);
  
  // Create subcategories for Electronics
  const phones = new Node("Phones");
  const computers = new Node("Computers");
  
  // Add subcategories to Electronics
  electronics.addChild(phones);
  electronics.addChild(computers);
  
  // Create subcategories for Clothing
  const men = new Node("Men");
  const women = new Node("Women");
  
  // Add subcategories to Clothing
  clothing.addChild(men);
  clothing.addChild(women);
  
  return store;
}

// Main function to run the program
function main() {
  // Build the tree
  const storeTree = buildStoreTree();
  
  // Print the tree structure
  console.log("Store Product Categories:");
  printTree(storeTree);
}

// Run the program
main();